src/sentry/api/endpoints/organization_details.py

```diff
@@ -522,15 +522,16 @@ def put(self, request: Request, organization) -> Response:
                     result = serializer.validated_data
 
                     if "slug" in changed_data:
-                        organization_mapping_service.create(
+                        organization_mapping_service.reserve_slug_for_organization(
```

Comment:
One thing that is going to fail is that `getsentry` depends on this method -- so you will see your getsentry tests fail by changing the name here.  There's a simple fix!

Implement a copy of the original method that calls the new one, so that the old name can remain.  Then rename this in the `getsentry` repo in a separate PR.  Then, after, you can remove the reference on this side.  It's a 3 stage rename.